core/livepeernode.go

```diff
@@ -473,6 +477,28 @@ func (n *LivepeerNode) StopEthServices() error {
 	return nil
 }
 
+func (n *LivepeerNode) GetClaimManager(job *ethTypes.Job) (eth.ClaimManager, error) {
+	n.claimMutex.Lock()
+	defer n.claimMutex.Unlock()
+	if job == nil {
+		glog.Error("Nil job")
+		return nil, fmt.Errorf("Nil job")
+	}
+	jobId := job.JobId.Int64()
+	// XXX we should clear entries after some period of inactivity
+	if cm, ok := n.ClaimManagers[jobId]; ok {
+		return cm, nil
+	}
+	// no claimmanager exists yet; check if we're assigned the job
+	if n.Eth == nil {
+		return nil, nil
```

Comment:
The plan was to return nil (instead of an error) to facilitate off-chain transcoding, since the `TranscodeAndBroadcast` function has an off-chain broadcasting feature that works if `n.Eth == nil`. However, the whole off-chain flow might be a little broken for now with the new networking scheme.

I'm not opposed to returning an error here, and ensuring that any off-chain check occurs before this `GetClaimManager` function is called. What do you think?

Reply:
Ah right so the ClaimManager is passed into `node.TranscodeAndBroadcast()` as `cm` which will assume it is running in off-chain mode if `cm == nil`. In that case, I think it is fine to stick with just returning nil for now then to follow the current assumptions for off-chain mode